### PR TITLE
[3.x] Add static analysis

### PR DESCRIPTION
**Pull request by jordyvanderhaegen**
This PR adds support for static analysis using PHPStan.
It's a first step towards making 3.x more type-safe.
Currently it uses PHPStan level 5, which should be fine for now.
Additionally, it includes several code changes, which I will outline briefly:

1. `LaravelLocalization@getURLFromRouteNameTranslated()`
A redundant if-check was performed while checking if `$locale` was a string.
This check was already performed earlier in this method.
    ```diff
    - if (\is_string($locale) && $this->translator->has($transKeyName, $locale))
    + if ($this->translator->has($transKeyName, $locale))
    ```
2. `LaravelLocalizationMiddlewareBase@except`
A null-check was performed on the `except` property, which was not assigned a value.
`$this->except = $this->except ?? config('laravellocalization.urlsIgnored', []);`
    ```diff
    - protected $except;
    + protected $except = null;
    ```

3. `LocaleCookieRedirect@handle()`
The default value passed to the cookie method is not allowed by the framework, which has `string|array|null` as a type annotation. We now use a null coalescing operator and assign the `false` value ourselves.
Whilst this wouldn't result in any issues, it's safe for when the framework decides to enforce parameter types.
    ```diff
    - $locale = $request->cookie('locale', false);
    + $locale = $request->cookie('locale') ?? false;
    ```

4. `LocaleSessionRedirect@handle()` & `LocaleCookieRedirect@handle()`
A safety-check was performed to avoid an undefined-index call on `$params[0]`.
However, the `explode` method that is used to assign`$params = explode('/', $request->path());` will always contain a first element. This check is therefor useless.
    ```diff
    - \count($params) > 0 && app('laravellocalization')->checkLocaleInSupportedLocales($params[0])
    + app('laravellocalization')->checkLocaleInSupportedLocales($params[0])
    ```

Other changes in this PR are type annotations which were not defined correctly, causing PHPStan to fail.
Additionally, once check fails for the `CustomTranslatorTest`, which is likely to change behaviour later on in 3.x.

### CODE REVIEW

File: src/Mcamara/LaravelLocalization/LaravelLocalization.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ class LaravelLocalization
     /**
      * Current locale.
      *
-     * @var string
+     * @var string|false
      */
     protected $currentLocale = false;
```
Comment (iwasherefirst2) [inline src/Mcamara/LaravelLocalization/LaravelLocalization.php:52]: How about remove comment and just add
```suggestion
    protected string|bool $currentLocale = false;
```
?

Comment (iwasherefirst2) [reply src/Mcamara/LaravelLocalization/LaravelLocalization.php:52]: I see you did that on other branch, nvm

**Comment by iwasherefirst2** [comment]
This is great!

**Comment by iwasherefirst2** [comment]
But why are there 50 commits? Most of then should be part of 3.x already? Could you cleanup those 50 commits?

**Comment by jordyvanderhaegen** [comment]
> But why are there 50 commits? Most of then should be part of 3.x already? Could you cleanup those 50 commits @jordyvanderhaegen ?

This branch was started from `3.x-inital-draft`, not from `3.x`. Hence why it includes the commits but not the changes.

**Pull request merged**